### DISCUSSION TITLE
Understanding problem with controllers with few RAM

### DISCUSSION THREAD

**Discussion by WolleKette** [micropython/micropython#11289] [General]
For a special project I need a microcontroller from STM. It is the first time for me to use STM controller. I orderd the NUCLEO L432KC board, which has quite few flash, but I don't care in this case. I have read that because of this, the controller doesn't provide a file system. Therefore I can't copy Python files to the board as usual. Micropython itself I have of course already installed.

I have no idea how to work with Micropython in such a case. I have read something about precompiled .mpy files. But I can't find any tutorials on how to work with them or how to get them on the board.

Please give me a hint. Many thanks for your help.

**Comment by dlech** [comment]
https://docs.micropython.org/en/latest/reference/manifest.html

**Comment by WolleKette** [comment]
A short link can be a great help. Thanks for this! Somehow I had ignored that until now. Stupid!

It took me a while to install the whole toolchain under macOS etc.. But now I'm able to compile Micropython with own modules for the NUCLEO-L432KC board and flash it to the board.
